### ISSUE TITLE
Add resource.Copy

### ISSUE THREAD

**Issue by regisphilibert** [gohugoio/hugo#9313]
Currently only some resource methods can take a "destination" argument. 

But many don't, including image transformation. 

It might also be useful to rename an asset without doing anything to it, especially when using `.GetRemote`:

```
{{ with resources.GetRemote "https://image.com/3093i49039i403i4903i403.jpg" }}
  {{ $img := resources.Copy . "directory/good-looking-name.jpg" }}
  <img src="{{ $img.RelPermalink }}" />
{{ end }}
```

**Comment by bep** [comment]
I suspect that we have discussed this before; my conclusion then would be that `resources.Copy` would be better (the old will still exist under its original name).

**Comment by regisphilibert** [comment]
Yes I could not find the conversation though.

**Comment by regisphilibert** [comment]
But then once you've copied it, are you suppose to "Get" it again from its new location?

**Comment by bep** [comment]
>Or the new asset is returned by the resources.Copy method...

Yes, you can start using it right away:

```
{{ $bar := resources.GetRemote "https://example.org/foo.jpg" | resources.Copy "bar.jpg" }}
<img src="{{ $bar.RelPermalink }}">
```

**Comment by regisphilibert** [comment]
Damn! How did I miss that! Thanks!

**Comment by jmooring** [comment]
What we have now:

```
{{ $options := dict "targetPath" "css/foo.css" }}
{{ $css := resources.Get "sass/style.scss" | resources.ToCSS $options }}

{{ $options := dict "targetPath" "js/foo.js" }}
{{ $js := resources.Get "js/script.js" | js.Build $options }}
````

To be somewhat consistent, we could just do the same for `.GetRemote`:

```
{{ $options := dict "targetPath" "img/foo.jpg" }}
{{ $img := resources.GetRemote "https://example.org/image.jpg" $options }}
```

There's an undocumented method to publish a resource, having the same effect as invoking `.Permalink` or `.RelPermalink`:

```
{{ $img := resources.GetRemote "https://example.org/image.jpg" }}
{{ $img.Publish }}
```

We could also create a `resources.Publish` method [#9303](https://github.com/gohugoio/hugo/issues/9303) which would take an optional dictionary:

```
{{ $options := dict "targetPath" "img/foo.jpg" }}
{{ $img := resources.GetRemote "https://example.org/image.jpg" | resources.Publish $options  }}
```

I'm not advocating any of these, but it's worth thinking about. Whatever we do I hope it's mostly consistent between global and page resources, regardless of media type (images, css, js, etc.).

**Comment by bep** [comment]
>To be somewhat consistent, we could just do the same for .GetRemote:

Yes, I have thought about that, and maybe we should do that, but there's a non-trivial implementation cost to add the same set of options to every `FooResource` function we have/add in the future. And I like the Unixy simplicity of this simple line:

```
{{ $bar := resources.GetRemote "https://example.org/foo.jpg" | resources.Copy "bar.jpg" }}
```

But I'm not totally 100% sure ... There may be other considerations as to why we maybe should add `targetPath` to at least the core funcs, e.g. `GetRemote/Get`.

**Comment by bep** [comment]
>{{ $img := resources.GetRemote "https://example.org/image.jpg" | resources.Publish $options  }}

Yea, that could maybe be a better idea; this would allow us to put more options in the future if needed.

**Comment by jmooring** [comment]
Related: <https://discourse.gohugo.io/t/using-getremote-on-image-puts-the-resource-in-root/36397>

**Comment by jmooring** [comment]
Duplicate of https://github.com/gohugoio/hugo/issues/7406

**Comment by regisphilibert** [comment]
> Yes, you can start using it right away:

Are you sure? I get this when I try your suggestion: 
```
at <resources>: can't evaluate field Copy in type interface {}
```

**Comment by jmooring** [comment]
Lost in translation...

regis asked: 

> But then once you've copied it, are you suppose to "Get" it again from its new location? Or the new asset is returned by the resources.Copy method...

bep answered:

> Yes, you can start using it right away:

bep didn't mean that the feature is available in the current release.

**Comment by regisphilibert** [comment]
> bep didn't mean that the feature is available in the current release.

I see, on `master` then?

**Comment by jmooring** [comment]
No. This is still in the proposal stage. Nothing has been implemented.

**Comment by stullesha** [comment]
This would help a lot! Any progress?
